var/spack/repos/builtin/packages/hermes/package.py

```diff
@@ -11,38 +11,136 @@ class Hermes(CMakePackage):
+        sha256="68e9a977c25c53dcab7d7f6ef0df96b2ba4a09a06aa7c4a490c67faa2a78f077",
+    )
+    version(
+        "0.9.5-beta",
+        url="https://github.com/HDFGroup/hermes/archive/refs/tags/v0.9.5-beta.tar.gz",
+        sha256="f48d15591a6596e8e54897362ec2591bc71e5de92933651f4768145e256336ca",
+    )
     version(
         "0.9.0-beta",
         url="https://github.com/HDFGroup/hermes/archive/refs/tags/v0.9.0-beta.tar.gz",
         sha256="abf258a52fa79729dfeb28559957abf8945f3ad37cadefb3bc685227c5f057a8",
     )
 
-    depends_on("c", type="build")  # generated
-    depends_on("cxx", type="build")  # generated
-
+    variant("adios", default=False, description="Build Adios tests")
+    variant("ares", default=False, description="Enable full libfabric install")
+    variant("compress", default=False, description="Enable compression")
+    variant("encrypt", default=False, description="Enable encryption")
+    variant("mpiio", default=True, description="Enable MPI I/O adapter")
+    variant("python", default=False, description="Build Python Wrapper")
+    variant("stdio", default=True, description="Enable STDIO adapter")
     variant("vfd", default=False, description="Enable HDF5 VFD")
+    variant("zmq", default=False, description="Build ZeroMQ tests")
+
+    depends_on("c", type="build")
+    depends_on("cxx", type="build")
+    depends_on("pkgconfig", type="build")
+    depends_on("libelf")
+
+    depends_on("hermes-shm@master+boost+cereal+mochi")
+
+    depends_on("hermes-shm+adios", when="+adios")
+    depends_on("hermes-shm+ares", when="+ares")
+    depends_on("hermes-shm+compress", when="+compress")
+    depends_on("hermes-shm+encrypt", when="+encrypt")
+    depends_on("hermes-shm+mpiio", when="+mpiio")
+    depends_on("hermes-shm+vfd", when="+vfd")
+    depends_on("hermes-shm+zmq", when="+zmq")
 
-    depends_on("mochi-thallium~cereal@0.8:")
-    depends_on("catch2")
-    depends_on("glpk")
-    depends_on("glog@0.4.0:")
-    depends_on("mpi")
-    depends_on("hdf5@1.14.0:", when="+vfd")
-    depends_on("yaml-cpp")
+    depends_on("py-jarvis-util", type="test")
+
+    depends_on("mpi", when="+mpiio")
+    conflicts("^[virtuals=mpi] nvhpc", when="+mpiio", msg="+mpio does not support nvhpc MPI")
 
     def cmake_args(self):
-        args = [
-            self.define("HERMES_RPC_THALLIUM", True),
-            self.define("HERMES_INSTALL_TESTS", True),
-            self.define("BUILD_TESTING", True),
-            self.define_from_variant("HERMES_ENABLE_VFD", "vfd"),
-        ]
+        args = []
+        if "+mpiio" in self.spec:
+            args.append("-DHERMES_ENABLE_MPIIO_ADAPTER=ON")
+            mpi_name = self.spec["mpi"].name
+            if mpi_name == "openmpi":
+                args.append("-DHERMES_OPENMPI=ON")
+            elif mpi_name == "mpich":
+                args.append("-DHERMES_MPICH=ON")
+            else:
+                raise InstallError("hermes+mpiio needs openmpi or mpich, got " + mpi_name)
```

Comment:
This now takes care of getting the selected virtual for `"mpi"` (as used in other recipes for `mpi` and `blas`) and raises an install error if another provider that is not supported is selected.